src/main/java/frc/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -32,32 +65,60 @@
           // Back right
           new Translation2d(-Dims.TRACKWIDTH_METERS / 2.0, -Dims.WHEELBASE_METERS / 2.0));
 
-  private final AHRS navx = new AHRS(SPI.Port.kMXP, (byte) 200); // NavX connected over MXP
+  /**
+   * The SwerveDriveOdometry class allows us to estimate our vehicle position over time using the
+   * power of :sparkles: math :sparkles:.
+   */
+  private final SwerveDriveOdometry swerveOdometry =
+      new SwerveDriveOdometry(kinematics, navx.getRotation2d());
+
+  /**
+   * Keeps track of the current estimated pose (x,y,theta) of the robot, as estimated by odometry.
+   */
+  private Pose2d robotPose = new Pose2d();
+
+  /** The current ChassisSpeeds goal for the drivetrain */
+  private ChassisSpeeds chassisSpeeds = new ChassisSpeeds(); // defaults to zeros
+
+  /** The mode of the drivebase subsystem */
+  public enum Modes {
+    DRIVE, // For autonomous operation, the DRIVE mode is used -- no need to create another
+    DEFENSE,
+  }
+
+  /** The current mode */
+  private Modes mode = Modes.DRIVE;
 
-  private final SwerveModule frontLeftModule;
-  private final SwerveModule frontRightModule;
-  private final SwerveModule backLeftModule;
-  private final SwerveModule backRightModule;
```

Comment:
Is there a reason to only store swerve modules in the array? Java object constructors return pointers, so there is no bugs inherent with having both. I think there are times where one is more ergonomic than the other. Thoughts?

Reply:
I didn't see a reason to keep too many members in scope and lead to potential bugs when it doesn't seem like we'd ever need to address modules individually, by name. What would be an example of where we'd see benefits from having these named aliases?